esp/esp/program/modules/handlers/schedulingcheckmodule.py

```diff
@@ -198,12 +198,29 @@ def _timeslot_dict(self, slot=lambda: 0):
                d[i] = slot()
           return d
 
-     #memoize the list of class sections in this program
+     #memoize the list of all class sections in this program
      def _all_class_sections(self):
           if self.listed_sections:
                return self.all_sections
           else:
                qs = self.p.sections()
+               #filter out non-approved classes
+               qs = qs.exclude(status__lte=0)
+               qs = qs.exclude(resourceassignment__isnull=True)
+               #filter out lunch
+               qs = qs.exclude(parent_class__category__category=u'Lunch')
+               qs = qs.select_related('parent_class', 'parent_class__parent_program', 'parent_class__category')
+               qs = qs.prefetch_related('meeting_times', 'resourceassignment_set', 'resourceassignment_set__resource', 'parent_class__teachers')
+               self.all_sections = list(qs)
```

Comment:
I think the "memoize" comment refers to this; when `_all_class_sections` is called, the result is stored in `self.all_sections` so that it doesn't need to be recomputed next time. But this doesn't work if the two functions both use the same variable.